src/java.base/share/classes/java/util/concurrent/ForkJoinPool.java

```diff
@@ -2477,9 +2428,8 @@ else if (mq == 1)
                                 }
                                 break;
                             }
-                            if (U.compareAndSetReference(
-                                    a, slotOffset(k), t, null)) {
-                                q.updateBase(nb);
+                            if (U.compareAndSetReference(a, k, t, null)) {
+                                q.base = nb;
                                 w.source = j;
```

Comment:
Might be worth commenting the volatile-write piggybacking which allows to get away with not calling updateBase above? 🤔